### ISSUE TITLE
Waheguru Ji Replicate website wahegurulive.com

### ISSUE THREAD

**Issue by cingh-jasdeep** [gpmkc-moga/waheguru-live-com#1]
Waheguru Ji Ka Khalsa
Waheguru Ji Ki Fateh ji

replicating website from 
wahegurulive.com

Wordpress
with 
Nuxt, Nuxt Content, Forestry

**Comment by cingh-jasdeep** [comment]
Nuxt setup
Prettier setup
Forestry setup
Github pages setup

**Comment by cingh-jasdeep** [comment]
forestry remote admin

**Comment by cingh-jasdeep** [comment]
next need to figure out how to send one signal notification.

**Comment by cingh-jasdeep** [comment]
need to check home url forestry

**Comment by cingh-jasdeep** [comment]
> need to check home url forestry

works!

**Comment by cingh-jasdeep** [comment]
> next need to figure out how to send one signal notification.

can work with github actions
in progress here:
https://github.com/cingh-jasdeep/onesignal-github-action

**Comment by cingh-jasdeep** [comment]
onesignal_nuxt setup
https://github.com/nuxt-community/onesignal-module

**Comment by cingh-jasdeep** [comment]
need to add loading indicator

**Comment by cingh-jasdeep** [comment]
- push notification should focus on already opened tab, instead of opening new window
- should not display response, only if error found

**Comment by cingh-jasdeep** [comment]
- push notification only on hukumnama file change in master branch only

**Comment by cingh-jasdeep** [comment]
- [x] use curl instead of action to send notification
- [x] push notification should focus on already opened tab, instead of opening new window(need to be done at client side)
notificationClickHandlerMatch, https://documentation.onesignal.com/docs/web-push-sdk
- [x] should build pages first then send notification if possible , concurrency problem, but not big one

**Comment by cingh-jasdeep** [comment]
Waheguru Ji Ka Khalsa
Waheguru Ji Ki Fateh ji

Today worked on push notification
works fine
prefer to use curl as there are no dependencies in this
https://documentation.onesignal.com/reference/create-notification#send-to-a-specific-segment-or-all-subscribers---create-notification
```
curl --include \
     --request POST \
     --header "Content-Type: application/json; charset=utf-8" \
     --header "Authorization: Basic YOUR_REST_API_KEY" \
     --data-binary "{\"app_id\": \"YOUR_APP_ID\",
\"contents\": {\"en\": \"English Message\"},
\"headings\": {\"en\": \"English Title\"},
\"included_segments\": [\"Subscribed Users\"]}" \
     https://onesignal.com/api/v1/notifications
```

**Comment by cingh-jasdeep** [comment]
Waheguru Ji Ka Khalsa
Waheguru Ji Ki Fateh ji

Today completed the Hukumnama notifications

Now need to design website, add events, add api integration with whatsapp catalog if possible

1. first need to design static layout
2. add programs
3. add products catalog with whatsapp if possible.
4. should add photo to notification if possible.

**Comment by cingh-jasdeep** [comment]
Waheguru Ji
Started design static layout website
- [x] need to upgrade tailwind nuxt plugin

**Comment by cingh-jasdeep** [comment]
Waheguru Ji Ka Khalsa
Waheguru Ji Ki Fateh ji

By Waheguru's grace
Today I worked on
1. Top Navbar

completed responsive 
top navbar

next to add slider
https://github.com/lukaszflorczak/vue-agile/tree/legacy/vue-v2
https://warpcgd.github.io/vue-concise-slider/

Waheguru Ji 🙏

**Comment by cingh-jasdeep** [comment]
Waheguru Ji Ka Khalsa
Waheguru Ji Ki Fateh ji

By Waheguru's grace 
Today I worked on

adding a slider for the home page
- [x] added input on forestry admin
- [x] add a slider plugin (which would work on SSR)

Found:
https://github.com/gs-shop/vue-slick-carousel

**Comment by cingh-jasdeep** [comment]
Waheguru Ji Ka Khalsa
Waheguru Ji Ki Fateh ji
By Waheguru's grace
Today I worked on

- complete home slider for responsive layout
- fixed home screen logo bug

pending 
- rest of static website
- add programs
- add products catalog with whatsapp if possible.
- should add photo to notification if possible.
   - fix onesignal, push notification should focus on already opened tab, instead of opening new window(need to be done at client side)
notificationClickHandlerMatch, https://documentation.onesignal.com/docs/web-push-sdk

**Comment by cingh-jasdeep** [comment]
Bug:
close menu on navigation on mobile

**Comment by cingh-jasdeep** [comment]
> Bug: close menu on navigation on mobile

fixed

**Comment by cingh-jasdeep** [comment]
Waheguru Ji Ka Khalsa
Waheguru Ji Ki Fateh 

By Waheguru Ji's Grace
Today got most of Hukumnama module finished
just push notif type left 
next steps give below

-add sticky header,after hukumnama
-add smagams block, later after hukumnama
-add home background, later after hukumnama


before final deployment of hukumnama need to test mobile
mobile should work
but data type should be mentioned

need to add push notification type to work on mobile

link
https://github.com/gpmkc-moga/gpmkc_mobile_app/blob/4228361ffc1ffedd58cef44da41133e0f6835130/lib/main.dart#L45
check in wordrpess also

seo need to add in index new
_in  process_



**BUG**
why subscribe not showind first stime check
yes this happening on mobile when 

diagnosis is that it works when land direct but not when coming from route
one possible solution is to provide your own custom code.

**Comment by cingh-jasdeep** [comment]
Waheguru Ji Ka Khalsa
Waheguru Ji Ki fateh ji

By Waheguru Ji's grace
Today worked on
- [Programs section](https://github.com/gpmkc-moga/gpmkc-moga.github.io/commit/efbc3165c2654323af36da607bc011f73ea8b083
)

- [Sticky header, fix bug Scrollbar shift content,](https://github.com/gpmkc-moga/gpmkc-moga.github.io/commit/bf168f2c36dc3f696215d2de07978fe2217b5c24
)

- OneSignal Subscribe button,
[add custom link prompt for nuxt from docs](https://github.com/gpmkc-moga/gpmkc-moga.github.io/commit/01cfc3d4b91eb5c70dcdf7aa821675589b30f7b2)
added gist for this!

- [onesignal-subscription-button.vue](https://gist.github.com/cingh-jasdeep/f9b7787e15899c746463c406bdc682c4)


Next steps
- add smagams block, almost complete, need "view all"
- add home background tomb
- add animations
https://zacheryng.com/nuxt-with-animated-on-scroll/
- add 404 page, with sitemap

**Comment by cingh-jasdeep** [comment]
**Bug**
Frame over menu

![Screenshot_20220515-175352.png](https://user-images.githubusercontent.com/5167015/168472747-7e63a55a-0f03-42d9-a8d8-60907cd08ee5.png)

**Comment by cingh-jasdeep** [comment]
Waheguru Ji Ka Khalsa
Waheguru Ji Ki Fateh ji

By Maalik's grace
today I worked on

[update: workflows](https://github.com/gpmkc-moga/gpmkc-moga.github.io/commit/0c1a5dc62cbd994febfe662fbee07a109bd952a7) 

- add concurrency groups for single execution
- add delay of 3minutes for hukumnama notification 

[fix nav-top, home-slider z-index, bg issues](https://github.com/gpmkc-moga/gpmkc-moga.github.io/commit/0bf02a3993b83303ee44613b78d5334ca3990bc2)

[Various more](https://github.com/gpmkc-moga/gpmkc-moga.github.io/commit/43ef6648219b078de8653146b63f8bcb7343ff3a) 
- add custom colors tailwind
- add poster to program-block, untested
- add view all button, rename program grid,
bug fix sort by date
- add tomb post component with common tomb styling for pages
- use tomb post in  today hukumnama page
- add programs page for all programs, incomplete

[more](https://github.com/gpmkc-moga/gpmkc-moga.github.io/commit/b265e349b92b3f1f1355ceb64c07e34373e338a6) 
- fix programs grid home
- add programs page
- add poster functionality
- update home page with new

[- add bhai sahib photo: home new](https://github.com/gpmkc-moga/gpmkc-moga.github.io/commit/80cc38861c0305ebeb6dbeca8a0a05d092d3ce42)


Shall continue with developing home page bit by bit and then add rest of pages
Waheguru Ji Ka Khalsa
Waheguru Ji Ki Fateh Ji

**Comment by cingh-jasdeep** [comment]
Waheguru Ji Khalsa
Waheguru Ji Ki Fateh ji

By Waheguru Ji's grace today I worked on

[-add program grid, leader section in home page](https://github.com/gpmkc-moga/gpmkc-moga.github.io/commit/a1a35ae6f1df602e860476027eeba32200df6df7)

[- add courses section home page](https://github.com/gpmkc-moga/gpmkc-moga.github.io/commit/e0be6c8177ba601d8ba14edeab8be8333e7f5d99)

[add home page section- we teach](https://github.com/gpmkc-moga/gpmkc-moga.github.io/commit/0d6c05c833d0bb68a5a62a2dd3077ef8a30dcb42)

[add extra classes section](https://github.com/gpmkc-moga/gpmkc-moga.github.io/commit/c0de7d10b069b53cb25421e2a02b07fd3f2cdfe2) 
- update alt text on images

[add live radio, video links to home_new](https://github.com/gpmkc-moga/gpmkc-moga.github.io/commit/b0736258064bca32351d47079e124f834790295d)

[- add basic radio player](https://github.com/gpmkc-moga/gpmkc-moga.github.io/commit/7e3b94aed63d7e21d487ac86f2cb01d1eb85df33) 
- add live video page
- add live video radio on home page


Tommorow I shall improve on radio player, make it little more UI friendly,
test it,
then About Page, Gallery, Store and Contact
basically other pages in order
also, footer needs to be updated
other tasks at end
-add animations, put in end, can check easily that way,
https://zacheryng.com/nuxt-with-animated-on-scroll/
-add 404 page, with sitemap, can add in end, can check easily if got all links that way
-maybe add a site-wide popup to open up at first page launch, not at subsequent
-that too after some interaction time, or whatever best metrics for it


Waheguru Ji Ka Khalsa
Waheugru Ji Ki Fateh ji!

**Comment by cingh-jasdeep** [comment]
Waheguru Ji Ka Khalsa
Waheguru Ji Ki Fateh ji

By Waheguru Ji's grace today I worked on
- [add radio page](https://github.com/gpmkc-moga/gpmkc-moga.github.io/commit/d05ed0f092bc76ea5dac20463f1ba26613e209cb)
- [add about page](https://github.com/gpmkc-moga/gpmkc-moga.github.io/commit/0d32630fb1471b9a1b45a9380109daff66b455ba)
- [finish about and gallery pages](https://github.com/gpmkc-moga/gpmkc-moga.github.io/commit/920569c399c4c7db877cfbe53d5408647dc80907)
- [add contact us page](https://github.com/gpmkc-moga/gpmkc-moga.github.io/commit/3da4df4148bf6f5fe1eab3dab552caa78f6c660a)
- add store page with just whatsapp link


other tasks left
footer needs to be created
- add animations, put in end, can check easily that way,
https://zacheryng.com/nuxt-with-animated-on-scroll/
- add 404 page, with sitemap, and manual links, https://sitemap.nuxtjs.org/, 
can add in end, can check easily if got all links that way

- maybe add a site-wide popup to open up at first page launch, not at subsequent
-t hat too after some interaction time, or whatever best metrics for i

**Comment by cingh-jasdeep** [comment]
Waheguru Ji Ka Khalsa
Waheguru Ji Ki Fateh ji

By Waheguru Ji's grace

Today I worked on 
- [add footer, 404 page and title for pages](https://github.com/gpmkc-moga/gpmkc-moga.github.io/commit/e55fa19fa087bd9e6460b38c5d51ce6c8f466fcb)
- [add animations](https://github.com/gpmkc-moga/gpmkc-moga.github.io/commit/40e66bddfdbf07d9663fce7c0f32a3434b239127)
- [update og tags](https://github.com/gpmkc-moga/gpmkc-moga.github.io/commit/708b509f1d53f757a85200b1adba5ecc99daf857)
- [try bug fix aos](https://github.com/gpmkc-moga/gpmkc-moga.github.io/commit/ae4db581ce429cf54e22af2c95f5c33cb5a424f0)
- [fix layout transition problem](https://github.com/gpmkc-moga/gpmkc-moga.github.io/commit/7a98ca7884e625c32a4ec1c90a27a38f09542f13)

site is almost complete
unless need 
- modal

- Also need access to custom domain 
to configure for push notifications
- need to check how to change type data of notification for mobile app to open correctly
- would be part of testing
